client/components/plans/plan-actions/index.jsx

```diff
@@ -189,14 +229,14 @@ module.exports = React.createClass( {
 			);
 		}
 
-		if ( isFreePlan( this.props.plan ) ) {
+		if ( this.shouldOfferFreeTrial() ) {
 			return (
-				<div onClick={ this.handleAddToCart.bind( null, null, 'button' ) } className={ classes } />
+				<div onClick={ this.handleClickStartTrial } className={ classes } />
```

Comment:
We should avoid using bind in render() because it creates a new function (slowly), making it necessary to update the DOM with every branch in the render tree containing a bind. I think that checking the event to avoid this was a reasonable fix, but this file already used bind like this and I think we can update it elsewhere.